### PR TITLE
Fix missing remote read spans

### PR DESCRIPTION
**Pull request by csmarchbanks**
The remote read client needs to use the nethttp.Transport wrapper in
order for spans to be instrumented properly.

Fixes: #7888 

cc: @roidelapluie and @bwplotka (who I think is on vacation)

### CODE REVIEW

**Comment by roidelapluie** [comment]
Do we have enough tags to the span?

**Comment by csmarchbanks** [comment]
> Do we have enough tags to the span?

Hmm? I don't understand what you are asking, could you rephrase your question? This change will restore the behavior that was present before the remote client was broken into separate read/write clients in https://github.com/prometheus/prometheus/pull/7059, but not add or remove any tags on the created spans.

**Comment by roidelapluie** [comment]
I was looking for nethttp.OperationName("Remote Read") but that is covered elsewhere.
